File: dsa04_LinkedList/node.py
class _Node:
    __slots__ = '_element', '_next'

    def __init__(self, element, n=None):
        self._element = element
        self._next = n


if __name__ == '__main__':
    n = _Node(10)
    print(n._element)
    n2 = _Node(20)
    n._next = n2
    print(n._element,n._next, n._next._element)

File: dsa04_LinkedList/node.go
package main

import (
	"fmt"
)

type Node struct {
	val  int
	next *Node
}

func main() {
	fmt.Println("----Node----")
	x := Node{30, nil}
	y := Node{20, &x}
	z := Node{10, &y}

	temp := &z

	for temp != nil {
		fmt.Println(temp.val, "->")

		temp = temp.next
	}
}
